plugin/storage/integration/kafka_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"bytes"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/jaeger-lib/metrics"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/config"
+	"github.com/jaegertracing/jaeger/pkg/testutils"
+	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+const defaultLocalKafkaBroker = "127.0.0.1:9092"
+
+type KafkaIntegrationTestSuite struct {
+	StorageIntegration
+}
+
+func (s *KafkaIntegrationTestSuite) initialize() error {
+	logger, _ := testutils.NewLogger()
+	s.logger = logger
+	topic := "jaeger-kafka-integration-test-" + strconv.FormatInt(time.Now().UnixNano(), 10)
+
+	f := kafka.NewFactory()
+	v, command := config.Viperize(f.AddFlags)
+	command.ParseFlags([]string{"--kafka.topic=" + topic, "--kafka.brokers=" + defaultLocalKafkaBroker})
+	f.InitFromViper(v)
+	err := f.Initialize(metrics.NullFactory, s.logger)
+	if err != nil {
+		return err
+	}
+	spanWriter, err := f.CreateSpanWriter()
+	if err != nil {
+		return err
+	}
+	spanReader, err := createSpanReader(topic)
+	if err != nil {
+		return err
+	}
+	s.SpanWriter = spanWriter
+	s.SpanReader = spanReader
+	s.Refresh = func() error { return nil }
+	s.CleanUp = func() error { return nil }
```

Comment:
Wouldn't there be overlap of testing data without clean up? Why not use a unique topic name between tests?

Reply:
I appended nanoseconds in Unix time when generating the topic.

Reply:
But you only do it once, not for each test.